src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -4,57 +4,103 @@ package freechips.rocketchip.diplomacy
+ *
+ * */
 abstract class LazyModule()(implicit val p: Parameters)
 {
+  /** contains sub-[[LazyModule]]s, can be accessed by [[getChildren]]. */
   protected[diplomacy] var children = List[LazyModule]()
+  /** contains [[BaseNode]] of this this instance. */
   protected[diplomacy] var nodes = List[BaseNode]()
+  /** stores [[SourceInfo]] of this instance, after instantiated, singleton of [[LazyModule]] will set it to right value. */
   protected[diplomacy] var info: SourceInfo = UnlocatableSourceInfo
-  protected[diplomacy] val parent = LazyModule.scope
+  /** father of this LazyModule, if this instance is top of hierarchy, should be [[None]]. */
+  protected[diplomacy] val parent: Option[LazyModule] = LazyModule.scope
 
-  // code snippets from 'InModuleBody' injection
+  /** code snippets from [[InModuleBody]] injection. */
   protected[diplomacy] var inModuleBody = List[() => Unit]()
 
+  /** chain of parents, [[parent]] on left, top on right. */
   def parents: Seq[LazyModule] = parent match {
     case None => Nil
     case Some(x) => x +: x.parents
   }
 
+  /** [[LazyModule.scope]] stack push. */
   LazyModule.scope = Some(this)
+  /** inject this instance to [[parent]] if this is not top. */
   parent.foreach(p => p.children = this :: p.children)
 
-  // suggestedName accumulates Some(names), taking the final one. Nones are ignored.
+  /** accumulates Some(names), taking the final one. Nones are ignored. */
   private var suggestedNameVar: Option[String] = None
+
+  /** override Verilog module name of [[LazyModuleImpLike]] module.
+    * @param x suggested name to [[LazyModuleImpLike]] module.
+    * */
   def suggestName(x: String): this.type = suggestName(Some(x))
   def suggestName(x: Option[String]): this.type = {
     x.foreach { n => suggestedNameVar = Some(n) }
     this
   }
 
+  /** helper of [[className]]. */
   private def findClassName(c: Class[_]): String = {
     val n = c.getName.split('.').last
     if (n.contains('$')) findClassName(c.getSuperclass) else n
   }
 
-  lazy val className = findClassName(getClass)
-  lazy val suggestedName = suggestedNameVar.getOrElse(className)
-  lazy val desiredName = className // + hashcode?
+  /** get the class name of this instance. */
+  lazy val className: String = findClassName(getClass)
+  /** used by [[LazyModuleImpLike.suggestName]]. */
+  lazy val suggestedName: String = suggestedNameVar.getOrElse(className)
+  /** used by [[LazyModuleImpLike.desiredName]]. */
+  lazy val desiredName: String = className // + hashcode?
 
+  /** @return instance name. */
   def name = suggestedName // className + suggestedName ++ hashcode ?
+  /** @return line to define this instance. */
   def line = sourceLine(info)
 
-  // Accessing these names can only be done after circuit elaboration!
-  lazy val moduleName = module.name // The final Verilog Module name
+  /** Accessing these names can only be done after circuit elaboration! */
+  /** The final Verilog Module name. */
```

Comment:
@richardxia what does "Chisel module name" mean?

Reply:
I was assuming that in Chisel there is some property that keeps track of the module name. Perhaps my understanding of this is incorrect? I still don't think that "Verilog module name" is the right wording, since Verilog is not a direct output of this process.

@jackkoenig, @azidar, could you offer a suggestion for what to write here? We could use someone with expertise in how module and instance names are represented in Chisel.

Reply:
resolved with
```
  // Accessing these names can only be done after circuit elaboration!
  /** Module name, used in GraphML. */
  lazy val moduleName = module.name
  /** Module hierarchy of this instance, used in GraphML. */
  lazy val pathName = module.pathName
  /** Instance name, used in GraphML. */
  lazy val instanceName = pathName.split('.').last
```

These variables only used in `GraphML` generation.